core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala

```diff
@@ -123,17 +122,12 @@ extends MemStore with StrictLogging {
                    cancelTask: Task[Unit]): CancelableFuture[Unit] = {
     val shard = getShardE(dataset, shardNum)
     combinedStream.map {
-                    case d: SomeData =>       shard.ingest(d)
+                    case d: SomeData =>       shard.ingest(d.records, d.offset, Some(flushSched))
```

Comment:
I don't think ingest() should involve itself in triggering the flush at all. 

I think a much cleaner separation would be like this:
  1) Remove the `FlushCommand`... that won't be necessary any more
  2) Code could be like this:

     case d: SomeData => shard.ingest(d)
                                           if (shard.checkTimeToFlush()) {
                                              Some(shard.prepareFlushGroup(group))
                                           } else None

Reply:
Yes. I left the FlushCommand in for now just to simplify the review.

Reply:
The reason why ingest triggers the flush is that it cannot absorb any more data until the flush has been prepared. If some stateful check must be invoked on the shard after ingest has been called, it's too late.

Reply:
This can be accomplished using the example I gave above.  It's not too late, as long as it happens before the next data is ingested.

Reply:
Another reason why I decided to have the shard trigger the flush is that it does so as it's iterating over the record container. If TimeSeriesMemStore needed to trigger the flush, then the record container would need to be iterated over twice (or more).

Reply:
We don't need to iterate over the record container to trigger the flush - there is only one ingestion time per recordContainer.  The trigger can be done before or after.

Reply:
True dat.

Reply:
...however, there's the monotonic clock check, which is adjusted per group. A monotonic clock per record container requires switching to 1 flush group, which I'm hoping to avoid.

Reply:
> there's the monotonic clock check, which is adjusted per group

How does this work again?  The per group check?

Reply:
The per group check is performed as sampled are extracted from the record container. which is heterogeneous.